### PR TITLE
Add rule for practice exercise `custom-set`

### LINKED ISSUE #49: Write analysis for practice exercise `custom-set`
The exercise is about writing a `Set`-like API and should not use any `Set` functions anywhere.

### PR DESCRIPTION
**Pull request by jiegillet**
Closes #49.
[Sister PR here](https://github.com/exercism/website-copy/pull/2284).
Pretty straightforward one using `Analyzer.functionCalls` :)

### CODE REVIEW

**Review by ceddlyburge — APPROVED** [review]
Looks good, and pleasingly simple :)

**Comment by jiegillet** [comment]
@exercism/maintainers-admin can I have an approval please? 💛

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
